src/components/Menu-new/buildDirectory.mjs

```diff
@@ -0,0 +1,178 @@
+// import fs from 'fs';
+// import path from 'path';
+
+// const metaObjectProperties = {
+//   title: 'title',
+//   description: 'description',
+//   platforms: 'platforms'
+// };
+
+/**
+ * @typedef PageNode
+ * @type {object}
+ * @property {string} title - Title of the page
+ * @property {string} description - Description of the page
+ * @property {booleam} isExternal - The node is an external link
+ * @property {string[]} platforms - The platforms that the pages apply to
+ * @property {string} route - The page's file path
+ * @property {PageNode[]} children - The children pages - optional
+ */
+
+// async function visitPage(directoryPath) {
+//   /** @type {PageNode} */
+//   const node = {
+//     route: path.basename(directoryPath)
+//   };
+
+//   const contents = fs.readdirSync(directoryPath);
+
+//   for (const item of contents) {
+//     if (
+//       path.extname(item).toLowerCase() === '.mdx' ||
+//       path.extname(item).toLowerCase() === ''
+//     ) {
+//       const itemPath = path.join(directoryPath, item);
+
+//       // Check if the item is a directory
+//       if (fs.statSync(itemPath).isDirectory()) {
+//         // Recurse
+//         visitPage(itemPath);
+
+//         if (contents.includes('index.mdx')) {
+//           // Root file for the directory
+//           // Get the frontmatter here
+//           const { meta } = await import(path.join(directoryPath, 'index.mdx'));
+
+//           for (const key of Object.keys(meta)) {
+//             node[key] = meta[key];
+//           }
+//         }
+//       } else {
+//         if (item === 'index.mdx') {
+//           // Skip this file since we already got the data when checking the directory
+//           continue;
+//         } else {
+//           // Child page
+//           const { meta } = await import(itemPath);
+
+//           for (const key of Object.keys(meta)) {
+//             node[key] = meta[key];
+//           }
+
+//           if (!Object.prototype.hasOwnProperty.call(node, 'children')) {
+//             node.children = [node];
+//           } else {
+//             node.children.push(node);
+//           }
+//         }
+//       }
+//     }
+//   }
+
+//   return node;
+// }
+
+// const result = await visitPage(
+//   '/Users/timngyn/Development/github/aws-amplify/docs/src/pages/'
+// );
+
+// console.log(result);
+
+export const directory = {
+  title: 'Amplify Docs',
+  description:
+    'Amplify documentation - Learn how to use Amplify to develop and deploy cloud-powered mobile and web apps.',
+  route: 'pages/index.tsx',
+  children: [
+    {
+      title: 'Overview',
+      description: `This is a description for the overview page.`,
```

Comment:
Wondering if we need this description property? (though I'm not as familiar with how our current directory file is used and if it is actually needed)

Reply:
For the menu, we probably don't need it. I was just copying+pasting your meta objects to build out this structure.

If we want to keep this object as simple as possible then I could definitely remove it